app/abilities/report.js

```diff
@@ -4,19 +4,26 @@ import { Ability } from "ember-can";
 export default Ability.extend({
   canEdit: computed(
     "user.{id,isSuperuser}",
-    "model.{user.id,user.supervisors,task.project.reviewers,verifiedBy,billed}",
+    "model.{user.id,user.supervisors,verifiedBy,billed,taskAssignees,projectAssignees,customerAssignees}",
     function() {
-      return (
+      const isEditable =
         this.get("user.isSuperuser") ||
         (!(this.get("model.verifiedBy.id") && this.get("model.billed")) &&
           (this.get("model.user.id") === this.get("user.id") ||
-            this.getWithDefault("model.user.supervisors", [])
+            (this.get("model.user.supervisors") ?? [])
               .mapBy("id")
-              .includes(this.get("user.id")) ||
-            this.getWithDefault("model.task.project.reviewers", [])
-              .mapBy("id")
-              .includes(this.get("user.id"))))
-      );
+              .includes(this.get("user.id"))));
+      const isReviewer =
+        (this.get("model.taskAssignees") ?? [])
+          .mapBy("user.id")
+          .includes(this.get("user.id")) ||
+        (this.get("model.projectAssignees") ?? [])
+          .mapBy("user.id")
+          .includes(this.get("user.id")) ||
+        (this.get("model.customerAssignees") ?? [])
+          .mapBy("user.id")
+          .includes(this.get("user.id"));
```

Comment:
Get rid of redundant code maybe like this:

```js
const reviewer = (this.get("model.taskAssignees") ?? [])
        .concat(
          this.get("model.projectAssignees") ?? [],
          this.get("model.customerAssignees") ?? []
        )
        .mapBy("user.id")
        .includes(this.get("user.id"));
```

Reply:
Works